packages/kit/src/runtime/server/page/actions.js

```diff
@@ -0,0 +1,153 @@
+import { error, json } from '../../../exports/index.js';
+import { normalize_error } from '../../../utils/error.js';
+import { negotiate } from '../../../utils/http.js';
+import { HttpError, Redirect, ValidationError } from '../../control.js';
+import { error_to_pojo } from '../utils.js';
+
+// Info: `enhance` action and `updateForm` live in `runtime/app/forms.js`
+
+/** @param {import('types').RequestEvent} event */
+export function is_action_json_request(event) {
+	const accept = negotiate(event.request.headers.get('accept') || 'text/html', [
+		'text/html',
+		'application/json'
+	]);
+
+	return (
+		accept === 'application/json' &&
+		event.request.method !== 'GET' &&
+		event.request.method !== 'HEAD'
+	);
+}
+
+/**
+ * @param {import('types').RequestEvent} event
+ * @param {import('types').SSROptions} options
+ * @param {import('types').SSRNode['server']} server
+ */
+export async function handle_action_json_request(event, options, server) {
+	const actions = server.actions;
+
+	if (!actions) {
+		maybe_throw_migration_error(server);
+		// TODO should this be a different error altogether?
+		return new Response('POST method not allowed. No actions exist for this page', {
+			status: 405,
+			headers: {
+				// https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405
+				// "The server must generate an Allow header field in a 405 status code response"
+				allow: 'GET'
```

Comment:
There's no GET request that accepts JSON under this url. Does it still count even if the GET only applies to a text/html request?

Reply:
based on my understanding of [RFC 9110](https://httpwg.org/specs/rfc9110.html#status.405), yes. it's not 100% clear but I'm pretty sure you're supposed to return a list of methods a resource supports regardless of `accept`